### PR TITLE
PERF: DataFrame.corr avoid copy

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx (Replace xxxx with the GitHub issue number)
- [ ] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [ ] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [ ] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Comment by phofl** [comment]
@jbrockmendel looks like a megre gone wrong?

**Comment by jbrockmendel** [comment]
Yes it does.  locally im looking at moving the fix to a lower level, not ready to push yet.

**Comment by jbrockmendel** [comment]
fix moved to a lower-level, merge problem fixed, green

File: pandas/core/internals/managers.py [hunk 1]
```diff
@@ -1689,17 +1689,29 @@ def as_array(
         -------
         arr : ndarray
         """
+        passed_nan = lib.is_float(na_value) and isna(na_value)
```
Comment (mroeschke) [inline pandas/core/internals/managers.py:1692]: `np.isnan`?

Comment (jbrockmendel) [reply pandas/core/internals/managers.py:1692]: causes mypy complaints

**Review by phofl — APPROVED** [review]
lgtm

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @jbrockmendel

**Pull request merged**